airflow/dags/unzip_and_validate_gtfs_schedule/unzip_gtfs_schedule.py

```diff
@@ -0,0 +1,205 @@
+    partition_names: ClassVar[List[str]] = GTFSScheduleFeedExtract.partition_names
+    ts: pendulum.DateTime
+    base64_url: str
+    zipfile_path: str
+    original_filename: str
+
+    # if you try to set table directly, you get an error because it "shadows a BaseModel attribute"
+    # so set as a property instead
+    @property
+    def table(self) -> str:
+        # only replace slashes so that this is a mostly GCS-filepath-safe string
+        # if we encounter something else, we will address: https://cloud.google.com/storage/docs/naming-objects
+        return self.original_filename.replace("/", "__")
+
+    @property
+    def dt(self) -> pendulum.Date:
+        return self.ts.date()
+
+
+class GTFSScheduleFeedExtractUnzipOutcome(ProcessingOutcome):
+    zipfile_extract_path: str
+    zipfile_extract_md5hash: Optional[str]
+    zipfile_files: Optional[List[str]]
+    zipfile_dirs: Optional[List[str]]
+    extracted_files: Optional[List[str]]
+
+
+class ScheduleUnzipResult(PartitionedGCSArtifact):
+    bucket: ClassVar[str] = SCHEDULE_UNZIPPED_BUCKET
+    table: ClassVar[str] = "unzipping_results"
+    partition_names: ClassVar[List[str]] = ["dt"]
+    dt: pendulum.Date
+    outcomes: List[GTFSScheduleFeedExtractUnzipOutcome]
+
+    @property
+    def successes(self) -> List[GTFSScheduleFeedExtractUnzipOutcome]:
+        return [outcome for outcome in self.outcomes if outcome.success]
+
+    @property
+    def failures(self) -> List[GTFSScheduleFeedExtractUnzipOutcome]:
+        return [outcome for outcome in self.outcomes if not outcome.success]
+
+    def save(self, fs):
+        self.save_content(
+            fs=fs,
+            content="\n".join(o.json() for o in self.outcomes).encode(),
+            exclude={"outcomes"},
+        )
+
+
+def summarize_zip_contents(zip: zipfile.ZipFile, at: str = ""):
+    files = []
+    directories = []
+    for entry in zip.namelist():
+        if zipfile.Path(zip, at=entry).is_file():
+            files.append(entry)
+        if zipfile.Path(zip, at=entry).is_dir():
+            directories.append(entry)
+    logging.info(f"Found files: {files} and directories: {directories}")
+    return files, directories
+
+
+def process_feed_files(
+    fs, extract: GTFSScheduleFeedExtract, zip: zipfile.ZipFile, directories: List[str]
+):
+    zipfile_files = []
+    # more than one directory --> can't parse
+    if len(directories) > 1:
+        raise ValueError("Unparseable zip: Multiple directories found")
+    # directory + any other artifact at top level --> can't parse
+    if directories and len([item for item in zipfile.Path(zip, at="").iterdir()]) > 1:
+        raise ValueError("Unparseable zip: Directory and other artifacts at root level")
+    # if the only artifact at the root level of the zipfile is a directory
+    # we can treat the contents of that directory as the feed
+    if [item.is_dir() for item in zipfile.Path(zip, at="").iterdir()] == [True]:
```

Comment:
nit: I think these two `if`s could be condensed a bit, they seem to share some aspects